src/olympia/api/paginator.py

```diff
@@ -49,3 +52,36 @@ def page(self, number):
                 raise EmptyPage('That page contains no results')
 
         return page
+
+
+class CustomPageNumberPagination(PageNumberPagination):
+    """Custom pagination implementation to hook in our `ESPaginator`."""
+    page_size_query_param = 'page_size'
+
+    def paginate_queryset(self, queryset, request, view=None):
```

Comment:
Instead of re-implementing `paginate_queryset`, can't you just set `django_paginator_class` class property ? https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/pagination.py#L172